### DISCUSSION TITLE
Shot on the start of a generated speech

### DISCUSSION THREAD

**Discussion by allohamora** [deepgram/community#664] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - TTS

### Details

When I generate a speech with mulaw output and sample rate 8000 I receive an audio with a shot on the start

### If possible, please attach your code or paste it into the text box.
```ts
import { createClient } from "@deepgram/sdk";
import { Readable, once } from "node:stream";
import { ReadableStream } from 'node:stream/web';
import { createWriteStream } from 'node:fs';
import { join } from "node:path";

const DG_KEY = 'write your key here';
export const client = createClient(DG_KEY);

const OUTPUT_PATH = join(__dirname, 'output.raw');

const text = 'Hello, World!';

const main = async () => {
  const res = await client.speak.request(
    { text },
    {
      encoding: 'mulaw',
      sample_rate: 8000,
    },
  );

  const webStream = await res.getStream();
  if (!webStream) {
    throw new Error(`webStream is not found for text: ${text}`);
  }

  const readStream = Readable.fromWeb(webStream as ReadableStream);
  const writeStream = createWriteStream(OUTPUT_PATH);

  readStream.pipe(writeStream);

  await once(readStream, 'end');

  console.log(`output file in ${OUTPUT_PATH}`)
};

main();
```

### If possible, please attach an example audio file to reproduce the issue.

[output.raw.zip](https://github.com/deepgram/community/files/14758177/output.raw.zip)

```bash
ffplay -autoexit -f mulaw -ar 8000 output.raw 
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment, marked as answer]
I'll try out this script soon, but one thing to do a quick check for is - the audio file might be a containerized wav file, not raw mulaw (we have a "container=wav" or "container=none" parameter one could use to tweak this) - when interpreting wav files (containing mulaw) as raw audio, the wav header often generates a "blip" at the beginning of playback. Maybe that's it?

**Reply by allohamora** [reply]
Yes, with container: 'none' I don't have this "shot" at the start, but with it, I have bad-quality audio, which isn't acceptable as well

[output-container.zip](https://github.com/deepgram/community/files/14769560/output-container.zip)

**Reply by kholiavko-roman** [reply]
Hi @nikolawhallon, I wanted to check if you had an opportunity to review this issue?

**Reply by kholiavko-roman** [reply]
friendly ping @nikolawhallon

**Reply by nikolawhallon** [reply]
The originally attached output.raw file is not raw audio but is containerized:
```
$ hexdump -C output.raw | head
00000000  52 49 46 46 24 00 ff 7f  57 41 56 45 66 6d 74 20  |RIFF$...WAVEfmt |
00000010  10 00 00 00 07 00 01 00  40 1f 00 00 40 1f 00 00  |........@...@...|
00000020  01 00 08 00 64 61 74 61  00 00 ff 7f 72 fe f4 f5  |....data....r...|
00000030  f3 fd 7d 7e 7b 7e 77 f6  f3 7e 7a 72 f4 f6 72 75  |..}~{~w..~zr..ru|
00000040  72 fa ed 7f 70 fc 7f f6  76 76 f4 78 7b 78 6c ef  |r...p...vv.x{xl.|
00000050  f9 7c fb 6e fd f3 70 f7  7b 6b eb fb 76 ee 7b f5  |.|.n..p.{k..v.{.|
00000060  f6 70 fb fc 7c f3 72 79  ec f7 fa 78 6b 7b 7a 79  |.p..|.ry...xk{zy|
00000070  f5 fa fe f2 79 f8 fc 7e  74 fc fe f1 ef 7c ec fc  |....y..~t....|..|
00000080  6d 7d 69 7a ef 6b fb fa  7d e5 7e 67 f7 70 71 f9  |m}iz.k..}.~g.pq.|
00000090  6f fe ef 7b f5 f9 7c eb  6d 70 f4 6f ef ee 6c f9  |o..{..|.mp.o..l.|
```

(you can tell from the RIFF header) so the header at the beginning will result in some jolt/bump/shot of audio if trying to play back the file as if it were all raw audio (it will play back fine in a media player which expects these headers though). Checking out output-container.zip right now.

**Reply by nikolawhallon** [reply]
The output-none file in output-container.zip does not have a shot of audio at the beginning, it looks and plays back nicely:

![image](https://github.com/deepgram/community/assets/43757888/28b571b4-ddae-430f-98a4-7702a7a9f663)

**Reply by nikolawhallon** [reply]
> Yes, with container: 'none' I don't have this "shot" at the start, but with it, I have bad-quality audio, which isn't acceptable as well

If you try to play back a file with non-audio data in it (like a header), that part will sound very bad. But these containerized files can be played back by most media players natively, without specifying that they are raw audio (they aren't).

**Reply by kholiavko-roman** [reply]
Yes, you are right. Without `container: 'none'` audio plays without jolt/bump/shot at the beginning. But the general audio quality is different - it's a little worse. It's possible to hear the difference between those 2 files [output-container.zip](https://github.com/deepgram/community/files/14769560/output-container.zip) especially with `S` letter.  It seems like the sound "lisps" in this case.

Is it possible to fix? 

Thank you.

**Reply by nikolawhallon** [reply]
The model's output can have slight variations per request (they are non deterministic), is this observation a consistent one over multiple sets of examples? If so can more examples be provided?

**Comment by rperezbws** [comment]
In case this helps anyone, I changed the TTS model and it stoped making that shot sound between sentences. The model making those sounds was "aura-asteria-en".
